homeassistant/components/sensor/enphase_envoy.py

```diff
@@ -81,27 +81,23 @@ def icon(self):
 
     def update(self):
         """Get the energy production data from the Enphase Envoy."""
-        import envoy_reader
+        from envoy_reader import EnvoyReader
 
         if self._type == "production":
-            self._state = int(envoy_reader.production(self._ip_address))
+            self._state = EnvoyReader(self._ip_address).production()
```

Comment:
If we add a map between sensor type and `EnvoyReader` method name, by extending `SENSORS`, we can simplify this code block significantly, by using `getattr`.